Add OSD_THROTTLE_POS_AUTO_THR indicator

fiam opened the pull request:
Displays stick THR position when the user controls THR directly,
actual applied THR when the navigation system controls THR.

Fixes #2031

Requires updated font from https://github.com/iNavFlight/inav-configurator/pull/258 and support in the OSD tab at https://github.com/iNavFlight/inav-configurator/pull/261

fiam committed:
Add OSD_THROTTLE_POS_AUTO_THR indicator

Displays stick THR position when the user controls THR directly,
actual applied THR when the navigation system controls THR.

Fixes #2031

digitalentity commented on src/main/navigation/navigation.c:
```diff
@@ -2693,6 +2693,11 @@ rthState_e getStateOfForcedRTH(void)
 
+bool navigationIsControllingThrottle(void)
+{
+    return posControl.navState != NAV_STATE_IDLE;
```
This is not always true. For multirotors we have 2D poshold mode which does not control throttle.

It's better to check for result of a call to `navGetCurrentStateFlags()` and also mind fixed-wing:
```
navigationFSMStateFlags_t stateFlags = navGetCurrentStateFlags();
return (stateFlags & (NAV_CTL_ALT | NAV_CTL_EMERG | NAV_CTL_LAUNCH | NAV_CTL_LAND)) || (STATE(FIXED_WING) && (stateFlags & (NAV_CTL_POS)));
```

fiam committed:
Better detection for THR control from the NAV system

As requested by @digitalentity. Previous code was returning a
false positive on MC when only 2D position hold was active.
